cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -364,6 +364,11 @@ globalCommand commands = CommandUI {
       ,multiOption "nix"
         globalNix (\v flags -> flags { globalNix = v })
         [
+          optArg' "(True or False)" maybeStringToBoolFlag (\case
+            Flag True -> [Just "enable"]
+            Flag False -> [Just "disable"]
+            NoFlag -> [Just "disable"]) "" ["nix"]
+            "Nix integration: run commands through nix-shell if a 'shell.nix' file exists (default is False)",
```

Comment:
Could this trivially be expressed using `trueArg` or `boolOpt` or similar? If not trivially, the extra `maybeStringToBoolFlag` function is fine.

Reply:
@Mikolaj That part is to make the `nix: True|False` part work in the config file. The issue was that only `enable-nix` and `disable-nix` worked for both the command line arg and the config file. This makes it so that all three work while showing the correct help text. I tried a few different things and none of them satisfied all the things I wanted to do.

Reply:
I can't remember exactly why I didn't go with it, but I'll try again and let you know

Reply:
@Mikolaj looked at it again and this is what I found:
`boolOpt` creates two separate help text fields with the only difference being that "enable" or "disable" is prepended to the help text. This is what the original pr was addressing because that was what was being used there and creating bad help text.

`trueArg|falseArg` I can use this in conjunction with `boolOpt` to get the help text to work but they end up overriding each other in practice and the tests fail.

`optArg` is actually a more verbose version of `optArg'` so it saves characters and complexity to use `optArg'` over `optArg` (I assume this is why `optArg'` is used so much more in general)

Reply:
Thank you for the analysis. TIL. Once a second review arrives, let's ship it!